### ISSUE TITLE
Is there a rule against unneccessary quotes?

### ISSUE THREAD

**Issue by louh** [stylelint/stylelint#4284]
> What is the problem you're trying to solve?

Just found a CSS rule that slipped through our codebase:

```css
.thing {
   margin-top: '10px';
}
```

The value in this case does not need to be quoted.

> What solution would you like to see?

I did my best to see if an existing rule could solve this problem, but it seems all the quote-related rules are to enforce the type of quotations marks when they're used, or its presence in CSS where they're expected (e.g. font names, URLs, the content value, etc). I didn't seem to find anything to disallow unnecessary quotation marks.

Following the convention of grouping rules, we could call this `value-no-unnecessary-quotes`. If `true`, the example above would fail and would need to be fixed. If `false`, the example above would not fail and it would be ignored.

If there is no existing solution and this seems like a valuable addition, I'd be happy to work with you on this.

**Comment by evilebottnawi** [comment]
You can create own plugin for this, it is non standard css, so plugin is best solution https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/plugins.md

**Comment by hudochenkov** [comment]
You might want to use [stylelint-csstree-validator](https://github.com/csstree/stylelint-validator).
